File: py_validation/pardox/_bindings/gpu.py
def bind(lib, c_void_p, c_char_p, c_int32, c_double, c_longlong, c_size_t, c_int64, c_int16):
    # Cross-manager variant (4-arg)
    for fn, args, ret in [
        ('pardox_gpu_add', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_gpu_mul', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_gpu_sub', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_gpu_div', [c_void_p, c_char_p, c_void_p, c_char_p], c_void_p),
        ('pardox_gpu_abs', [c_void_p, c_char_p],                      c_void_p),
        ('pardox_gpu_exp', [c_void_p, c_char_p],                      c_void_p),
        ('pardox_gpu_log', [c_void_p, c_char_p],                      c_void_p),
        ('pardox_gpu_sqrt',[c_void_p, c_char_p],                      c_void_p),
    ]:
        if hasattr(lib, fn):
            getattr(lib, fn).argtypes = args
            getattr(lib, fn).restype  = ret

File: js_validation/pardox/src/_bindings/gpu.js
'use strict';

module.exports = function bindGpu(_lib) {
    _lib.pardox_gpu_add  = _lib.raw.func('void * pardox_gpu_add(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_gpu_sub  = _lib.raw.func('void * pardox_gpu_sub(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_gpu_mul  = _lib.raw.func('void * pardox_gpu_mul(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_gpu_div  = _lib.raw.func('void * pardox_gpu_div(void * mgr, const char * col_a, const char * col_b)');
    _lib.pardox_gpu_sqrt = _lib.raw.func('void * pardox_gpu_sqrt(void * mgr, const char * col)');
    _lib.pardox_gpu_exp  = _lib.raw.func('void * pardox_gpu_exp(void * mgr, const char * col)');
    _lib.pardox_gpu_log  = _lib.raw.func('void * pardox_gpu_log(void * mgr, const char * col)');
    _lib.pardox_gpu_abs  = _lib.raw.func('void * pardox_gpu_abs(void * mgr, const char * col)');
};
